src/strategies/mangrove-station-qv-scaled-to-mgv/index.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  getAllMembers,
+  fetchBadgeBalances,
+  fetchScores
+} from '../station-score-if-badge';
+import { getAllVestings } from '../dss-vest-unpaid';
+
+export const author = 'espendk';
+export const version = '1.0.0';
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const members = await getAllMembers(
+    network,
+    provider,
+    snapshot,
+    options.membershipERC721,
+    options.erc6551Registry,
+    options.erc6551Implementation,
+    options.erc6551Salt
+  );
+
+  await fetchBadgeBalances(
+    network,
+    provider,
+    snapshot,
+    members,
+    options.badgesERC1155,
+    options.activeBadgeId
+  );
+
+  // Keep only TBAs and members with a badge
+  for (const [, member] of members) {
+    member.TBAs = member.TBAs.filter((tba) => (tba.badgeBalance ?? 0) > 0);
+    if (member.TBAs.length === 0) {
+      members.delete(member.address);
+    }
+  }
+
+  await fetchScores(
+    network,
+    provider,
+    snapshot,
+    members,
+    options.activityScoreERC20,
+    options.activityScoreDecimals
+  );
+
+  // Aggregate membership scores and apply quadratic voting
+  const activeMemberScores = new Map<string, number>();
+  let totalScore = 0;
+  for (const [, member] of members) {
+    let score = 0;
+    for (const tba of member.TBAs) {
+      score += tba.score ?? 0;
+    }
+    score = Math.sqrt(score);
+    activeMemberScores.set(member.address, score);
+    totalScore += score;
+  }
+
+  // Get the circulating supply of the vesting token
+  const allVestings = await getAllVestings(
+    network,
+    provider,
+    snapshot,
+    options.dssVestAddress,
+    options.dssVestDecimals
+  );
```

Comment:
This is crossing the limit of 5 requests per strategy 🙈 you see some other way to do this???

https://github.com/snapshot-labs/snapshot-strategies#code

Reply:
Everything else looks good

Reply:
Uh oh. I don't see any other way to do this w/o resorting to some off-chain indexing :-/

Reply:
I think this request is fine as it is very light